Implement Clone Key action to AnimationPlayer

sofia285 opened the pull request:
Added "Clone Key" button to the AnimationPlayer which allows you to duplicate and bind keys. Added "Make Unique" button, which breaks the connection between cloned keys.

Co-authored-by: Miguel Coelho <miguel.pinheiro.coelho@tecnico.ulisboa.pt>

sofia285 committed:
Implemented clone key action to AnimationPlayer

Added "Clone Key" button to the AnimationPlayer which allows you to
duplicate and bind keys. Added "Make Unique" button, which breaks the
connection between cloned keys.
This feature allows a user to select a key and clone it to a track in a
different time but still having the same proprieties, even if you change
one of them. The "Make Unique" is useful to break the connection between
the keys, making them independent from one another.

Co-authored-by: Miguel Coelho <miguel.pinheiro.coelho@tecnico.ulisboa.pt>

sofia285 committed:
Implemented clone key action to AnimationPlayer

Added "Clone Key" button to the AnimationPlayer which allows you to duplicate and bind keys. Added "Make Unique" button, which breaks the
connection between cloned keys.
This feature allows a user to select a key and clone it to a track in a different time but still having the same proprieties, even if you
change one of them. The "Make Unique" button is useful to break the connection between the keys, making them independent from one another.

Co-authored-by: Miguel Coelho <miguel.pinheiro.coelho@tecnico.ulisboa.pt>

AThousandShips commented on doc/classes/Animation.xml:
```diff
@@ -424,6 +424,46 @@
+			<param index="2" name="track_idx" type="int" />
+			<param index="3" name="key_idx" type="int" />
+			<description>
+				Binds a key to a list of binded keys. Returns the index of the added key in the list.
```
```suggestion
				Binds a key to a list of bound keys. Returns the index of the added key in the list.
```
For all instances

AThousandShips commented on scene/resources/animation.h:
```diff
@@ -115,7 +115,17 @@ class Animation : public Resource {
 		virtual ~Track() {}
 	};
 
+	struct BindedKey {
```
```suggestion
	struct BoundKey {
```

AThousandShips commented on scene/resources/animation.h:
```diff
@@ -236,6 +246,7 @@ class Animation : public Resource {
 	};
 
 	Vector<Track *> tracks;
+	Vector<BindParent *> binded_keys;
```
```suggestion
	Vector<BindParent *> bound_keys;
```

sofia285 committed:
Implemented clone key action to AnimationPlayer

Added "Clone Key" button to the AnimationPlayer which allows you to duplicate and bind keys. Added "Make Unique" button,
which breaks the connection between cloned keys. This feature allows a user to select a key and clone it to a track in a
different time but still have the same proprieties, even if you change one of them. The "Make Unique" button is useful to
break the connection between the keys, making them independent from one another.

Co-authored-by: Miguel Coelho <miguel.pinheiro.coelho@tecnico.ulisboa.pt>

sofia285 commented on scene/resources/animation.h:
```diff
@@ -115,7 +115,17 @@ class Animation : public Resource {
+	};
+
+	struct BindParent {
+		Vector<BindedKey *> bindedChildren;
```
Thanks for the advice, just changed it!

sofia285 committed:
Implemet clone key action to AnimationPlayer

Added "Clone Key" button to the AnimationPlayer which allows you to
duplicate and bind keys. Added "Make Unique" button, which breaks the
connection between cloned keys. This feature allows a user to select a
key and clone it to a track in a different time but still have the same
proprieties, even if you change one of them. The "Make Unique" button is
useful to break the connection between the keys, making them independent
from one another.

Co-authored-by: Miguel Coelho <miguel.pinheiro.coelho@tecnico.ulisboa.pt>
